### PR TITLE
support `link_selections` with Dask DataFrames

### PR DESCRIPTION
**Pull request by jonmmease**
Adds support for using the new `link_selections` transformation on elements backed by Dask DataFrames.

There were two problems before:
 1. The `keep_index` and `compute` args were not being passed along to subexpressions (fixed in 
c99c0cf)
 2. The `digitize` and `isin` numpy functions are not implemented as Pandas or Dask Series methods, so Dask Series were getting converted into numpy arrays.  I added custom wrappers for these functions in e38f68f.

Closes https://github.com/pyviz/holoviews/issues/4034

TODO:
 - [x] Tests: We should add tests across all dim transformations to make sure they preserve dask Series as Series when appropriate.

### CODE REVIEW

File: holoviews/util/transform.py [hunk 1]
```diff
@@ -375,7 +412,15 @@ def apply(
...
             for arg in args:
                 if isinstance(arg, dim):
-                    arg = arg.apply(dataset, flat, expanded, ranges, all_values)
+                    arg = arg.apply(
+                        dataset,
+                        flat,
+                        expanded,
+                        ranges,
+                        all_values,
+                        keep_index,
```
Comment (philippjfr) [inline holoviews/util/transform.py]: Ah, think I missed this, which is why my attempts failed.

File: holoviews/util/transform.py [hunk 2]
```diff
@@ -106,6 +106,36 @@ def categorize(values, categories, default=None):
     return np.asarray(cats)
 
 
+def _maybe_map(numpy_fn):
```
Comment (philippjfr) [inline holoviews/util/transform.py]: Was wondering whether we should do this for any numpy function that's not implemented but I think for now sticking with just isin and digitize is safest.

**Comment by jonmmease** [comment]
Ok, all dim expressions are now tested against Dask DataFrames. I had to fix Dask support for a few of the operations along the way, but I think things are in pretty good shape now.

**Comment by philippjfr** [comment]
Amazing, thanks so much @jonmmease!

**Comment by philippjfr** [comment]
Some minor python 2 issues in the tests.

**Comment by jonmmease** [comment]
Python 2 issues fixed

**Pull request merged**